unlock-app/src/__tests__/services/web3Service.test.js

```diff
@@ -133,6 +133,26 @@ describe('Web3Service', () => {
       web3Service.connect(Object.assign({}, defaultState))
     })
 
+    it('should silently ignore requests to connect again to the same provider', done => {
```

Comment:
Why does this happen?

Reply:
So, there is an infinite loop when the `connect` method` triggers an error since the `middleware` will intercept all events and while we're not connected, will issue a new `connect` call.